src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -77,14 +74,154 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
     }
     log.info("Registered framework frameworkId=" + frameworkId.getValue());
   }
-
   @Override
   public void reregistered(SchedulerDriver driver, MasterInfo masterInfo) {
     log.info("Reregistered framework.");
   }
 
+  @Override
+  public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
+    log.info(String.format(
+        "Received status update for taskId=%s state=%s message='%s' stagingTasks.size=%d",
+        status.getTaskId().getValue(),
+        status.getState().toString(),
+        status.getMessage(),
+        liveState.getStagingTasksSize()));
+
+    if (!isStagingState(status)) {
+      liveState.removeStagingTask(status.getTaskId());
+    }
+
+    if (isTerminalState(status)) {
+      liveState.removeTask(status.getTaskId());
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status); // TODO(rubbish) could terminal/running get pushed to
+                                             // this
+
+      log.info(String.format("Current Acquisition Phase: %s", liveState
+          .getCurrentAcquisitionPhase().toString()));
+
+      switch (liveState.getCurrentAcquisitionPhase()) {
+        case JOURNAL_NODES :
```

Comment:
Let's first create three journal nodes here. When we create the two namenodes we can colocate them with two of the journal nodes. The logic wouldn't have to change too much and this would give us the benefit of a lower probability of failure for our JN quorum. I would be glad to discuss in more detail.